pkg/management/component.go

```diff
@@ -508,6 +512,41 @@ func (c *component) GetGroupsOfUser(ctx context.Context, realmName, userID strin
 	return groupsRep, nil
 }
 
+func (c *component) SetGroupsToUser(ctx context.Context, realmName, userID string, groupIDs []string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	var groupsKc, err = c.keycloakClient.GetGroupsOfUser(accessToken, realmName, userID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return err
+	}
+
+	var alreadyActiveGroups = make(map[string]bool)
+	for _, grp := range groupsKc {
+		if !validation.IsStringInSlice(groupIDs, *grp.Id) {
+			err = c.keycloakClient.DeleteGroupFromUser(accessToken, realmName, userID, *grp.Id)
+			if err != nil {
+				c.logger.Warn(ctx, "err", err.Error())
+				return err
+			}
+		} else {
+			alreadyActiveGroups[*grp.Id] = true
```

Comment:
Just a note as it has no functional impact at all: I think it is not really needed to have such array. all those elements will already be in groupsKc so you could use groupKc in the next loop instead of this new array right ?

Reply:
Already discussed together... This is not an array but a map so it will be slight more performant to check in a map rather than in a slice when executing the second loop.